File: trunk/HeartRateMonitor.Server/HeartRateMonitor.Math/Evaluation/RMSSD.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace HeartRateMonitor.Math.Evaluation
{
    public class RMSSD : IEvaluation<int>
    {
        public int Evaluate(SessionData session)
        {
            long total = 0;

            for (int i = 1, count = session.Intervals.Count; i < count; i++) {
                var now = session.Intervals[i];
                var before = session.Intervals[i - 1];
                total += (now - before) * (now - before);
            }

            return (int) System.Math.Sqrt(total / (session.Intervals.Count - 1));
        }
    }


}

File: CardioMood/Math/src/main/java/com/cardiomood/math/parameters/RMSSD.java
package com.cardiomood.math.parameters;

/**
 * Created by danon on 18.02.14.
 */
public class RMSSD extends AbstractSingleValue {


    @Override
    public String getName() {
        return "RMSSD";
    }

    @Override
    public double evaluate(double[] x, double[] y, int begin, int length) {
        double s = 0.0;
        for (int i=begin+1; i<length+begin && i<y.length; i++) {
            s += y[i]-y[i-1];
        }
        return Math.sqrt(s);
    }
}
